File: Swift/XWeather/XWeather/Domain/Wu/GeoLookup.swift
//
//  GeoLookup.swift
//  XWeather
//
//  Created by Colby Williams on 9/13/16.
//  Copyright © 2016 Colby Williams. All rights reserved.
//

import Foundation

public class GeoLookup : WuObject {

	public override var WuKey: String { get { return "geolookup" } }

	public var location: GeoLookupLocation?
}


public class WeatherStation : GeoLocation {

	public var icao: String?
}


public class PersonalWeatherStation : WeatherStation {

	public var neighborhood: String?
	public var id: String?
	public var distance_km: Int?
	public var distance_mi: Int?
}


public class NearbyWeatherStations {

	public var airport: Airport?
	public var pws: Pws?
}


public class Airport {

	public var station:[WeatherStation]?
}


public class Pws {

	public var station:[PersonalWeatherStation]?
}


public class GeoLookupLocation : GeoLocation {

	public var type: String?
	public var country_name: String?
	public var tz_short: String?
	public var tz_long: String?
	public var zip: String?
	public var magic: String?
	public var wmo: String?
	public var l: String?
	public var requesturl: String?
	public var wuiurl: String?
	public var nearby_weather_stations: NearbyWeatherStations?
}

File: XWeather/Shared/Domain/Wu/GeoLookup.cs
using System.Collections.Generic;

namespace XWeather.Domain
{
	public class GeoLookup : WuObject
	{
		public override string WuKey => "geolookup";

		public GeoLookupLocation location { get; set; }
	}


	public class WeatherStation : GeoLocation
	{
		public string icao { get; set; }
	}


	public class PersonalWeatherStation : WeatherStation
	{
		public string neighborhood { get; set; }
		public string id { get; set; }
		public int distance_km { get; set; }
		public int distance_mi { get; set; }
	}


	public class NearbyWeatherStations
	{
		public Airport airport { get; set; }
		public Pws pws { get; set; }
	}


	public class Airport
	{
		public List<WeatherStation> station { get; set; }
	}


	public class Pws
	{
		public List<PersonalWeatherStation> station { get; set; }
	}


	public class GeoLookupLocation : GeoLocation
	{
		public string type { get; set; }
		public string country_name { get; set; }
		public string tz_short { get; set; }
		public string tz_long { get; set; }
		public string zip { get; set; }
		public string magic { get; set; }
		public string wmo { get; set; }
		public string l { get; set; }
		public string requesturl { get; set; }
		public string wuiurl { get; set; }
		public NearbyWeatherStations nearby_weather_stations { get; set; }
	}
}